docs/contents/user-docs/irsa.md

```diff
@@ -31,8 +31,150 @@ to assume the IAM Role with reduced permissions.
+aws iam attach-role-policy \
+    --role-name "${ACK_EKS_IRSA_IAM_ROLE_NAME}" \
+    --policy-arn "$IAM_POLICY_ARN"
+```
+
+### 3. Associate an IAM role to service account**
+
+For detailed instructions, follow instructions at [Associate an IAM role to a service account ][5].
+
+#### 3(a) - If helm charts available on local file system
+
+Update `values.yaml` and set value for `aws.region`, `serviceAccount.annotations`.
+
+```
+# update variables as needed
+ACK_CONTROLLER_HELM_CHARTS_DIR=<directory containing helm chart for ACK service controller>
+AWS_SERVICE_NAME='elasticache'
+ACK_K8S_NAMESPACE=ack-system-test-helm
+ACK_K8S_DEPLOYMENT_NAME=ack-$AWS_SERVICE_NAME-controller-helm-test
+
+kubectl create namespace "$ACK_K8S_NAMESPACE"
+cd "$ACK_CONTROLLER_HELM_CHARTS_DIR"
+
+# dry run and view the resultant output
+helm install --debug --dry-run --namespace "$ACK_K8S_NAMESPACE" "$ACK_K8S_DEPLOYMENT_NAME" .
+# install on cluster
+helm install --namespace "$ACK_K8S_NAMESPACE" "$ACK_K8S_DEPLOYMENT_NAME" .
+```
+
+Verify that the service account has been created on cluster and that its annotation include IAM Role
+ (created during Step#2 above) arn:
+```
+kubectl describe serviceaccount/$ACK_K8S_SERVICE_ACCOUNT_NAME -n $ACK_K8S_NAMESPACE
+```
+
+#### 3(b) - If helm charts have already been installed on cluster without modifying `valyes.yaml`
+
+For example, if installation was done as:
+```
+AWS_SERVICE_NAME='elasticache'
+ACK_K8S_NAMESPACE=ack-system-test-helm
+ACK_K8S_DEPLOYMENT_NAME=ack-$AWS_SERVICE_NAME-controller-helm-test
+helm install --namespace $ACK_K8S_NAMESPACE ack-$AWS_SERVICE_NAME-controller $ACK_K8S_DEPLOYMENT_NAME
+```
+Then service account would already exist on the cluster; however its association with IAM Role would be pending.
+Verify it using:
+```
+kubectl describe serviceaccount/$ACK_K8S_SERVICE_ACCOUNT_NAME -n $ACK_K8S_NAMESPACE
+```
+Observe that the arn of IAM Role (created during Step#2 above) is not set as annotation for the service account.
+
+To associate an IAM role to service account:
+```
+# annotate service account with service role arn.
+ISRA_ROLE_ARN=<role arn>
+kubectl annotate serviceaccount -n $ACK_K8S_NAMESPACE $ACK_K8S_SERVICE_ACCOUNT_NAME $ISRA_ROLE_ARN
+```
+
+Update aws region to use in the controller, if not done already:
+```
+# update desired AWS region. example: us-east-1
+AWS_REGION=<aws region id>
+kubectl -n $ACK_K8S_NAMESPACE set env deployment/$ACK_K8S_DEPLOYMENT_NAME \
+    AWS_REGION="$AWS_ACCOUNT_ID"
+```
+
+### Verify
+Describe one of the pods and verify that the `AWS_WEB_IDENTITY_TOKEN_FILE` and `AWS_ROLE_ARN` environment variables exist.
+```
+kubectl get pods -A
+kubectl exec -n kube-system aws-node-<9rgzw> env | grep AWS
+```
+verify the output, example:
+```
+AWS_VPC_K8S_CNI_LOGLEVEL=DEBUG
+AWS_ROLE_ARN=arn:aws:iam::<AWS_ACCOUNT_ID>:role/<IAM_ROLE_NAME>
+AWS_WEB_IDENTITY_TOKEN_FILE=/var/run/secrets/eks.amazonaws.com/serviceaccount/token
+```
```

Comment:
Eventually I'd love to convert the above instructions into a Bash script that can simplify the experience for users. Eventually :)